File: src/agg.c
// Made with GNU/Linux
#include <stdio.h>
#include <stdbool.h>
#include <time.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
int main(void) {
  srand(time(NULL));

  char *username = getenv("USER");
  printf ("Hello %s!\nThis is a game written for AgnoxGD.\nWould you like to see a tutorial\nType 1 for no, or 0 for yes.\n", username);
  int tempInput;
  bool tutorial;
  if (scanf ("%d", &tempInput) == 1) {
    tutorial = tempInput; }
  if (tutorial) {
    printf ("OK! Lets continue.\n");
  } else {
    printf ("Welcome to Agg! This game tests your math skills against the clock.\n\nHow to Play:\n\n1.  Set Your Playtime: First, you'll decide how long you want to play by entering a duration in seconds.\n2.  Solve Math Problems: Once the game starts, you'll be presented with a series of random math problems: addition, division, subtraction, and multiplication. Your goal is to solve as many as you can before time runs out.\n3.  Enter Your Answer: After each problem, type your answer and press Enter. The game will immediately tell you if you're correct or incorrect.\n4.  Rack Up Points: For every correct answer, you'll earn a point. Your score will be tallied at the end.\n5.  Time's Up! The game ends automatically when your chosen playtime runs out. Good luck, and have fun!\n"); }

  int accuracytype;
  printf ("Would you like your accuracy as a float or as an int? Type 0 for float or 1 for int.\n");
  scanf ("%d", &accuracytype);

  int playtime;
  printf ("\nHow long would you like to play for? Answer in seconds.\n");
  scanf ("%d", &playtime);

  int number_length;
  printf ("Please enter maximum possible number\n");
  scanf ("%d", & number_length);

  time_t unix_time;
  unix_time = time(NULL);
  int end_time = unix_time + playtime;

  int score = 0;
  int questioncount = 0;

  while (unix_time < end_time) {
    float num1 = rand() % number_length + 1;
    float num2 = rand() % number_length + 1;
    int type = rand() % 4 + 1;
    float result;
    float answer;
    char entered[20];

    // DEBUG
    //printf("%g, %g and %d\n", num1, num2, type);

    if (type == 1) {
      printf ("Add %g and %g\n", num1, num2);
      result = num1 + num2;
    } else if (type == 2) {
      printf ("Divide %g by %g\n", num1, num2);
      result = num1 / num2;
    } else if (type == 3) {
      printf ("Subtract %g from %g\n", num2, num1);
      result = num1 - num2;
    } else if (type == 4) {
      printf ("Multiply %g by %g\n", num1, num2);
      result = num1 * num2;
    }
    unix_time = time(NULL);
    questioncount++;

    scanf ("%s", entered);
    int valid = 1;
    for (int i = 0; i < strlen(entered); i++) {
      switch (entered [i]) {
        case '0':
        break;
        case '1':
        break;
        case '2':
        break;
        case '3':
        break;
        case '4':
        break;
        case '5':
        break;
        case '6':
        break;
        case '7':
        break;
        case '8':
        break;
        case '9':
        break;
        case '.':
        break;
        default:
        valid = 0;
      }
    }

    if (valid == 1) {
      sscanf (entered, "%g", &answer);
    } else {
      printf ("Error! An invalid character was entered.\n");
      exit (1);
    }

    if (result == answer) {
      printf ("Correct!\n");
      score++;
    } else {
      printf ("Incorrect! The correct answer is %g.\n", result);
    }

    // DEBUG
   // printf ("Result is %g\n", result);
  }

  float accuracy;

  accuracy = (float)score / questioncount * 100;

  printf ("Your score is %d!\n", score);
  if (score >= 1) {
    printf ("Great job, %s!\n", username);
  } else {
    printf ("Try again!, %s.\n", username);
  }
  if (accuracytype == 1) {
    printf ("You got an accuracy of %d%%!\n", (int)accuracy);
  } else {
    printf ("You got an accuracy of %g%%!\n", accuracy);
  }
  return 0;
}

File: src/agg.js
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Agg Math Game</title>
    <style>
        body {
            font-family: monospace;
            background: #000;
            color: #0f0;
            padding: 20px;
            line-height: 1.4;
        }
        .container {
            max-width: 800px;
            margin: 0 auto;
        }
        input {
            background: #000;
            color: #0f0;
            border: 1px solid #0f0;
            padding: 5px;
            font-family: monospace;
            margin: 5px 0;
        }
        button {
            background: #000;
            color: #0f0;
            border: 1px solid #0f0;
            padding: 10px 20px;
            font-family: monospace;
            cursor: pointer;
            margin: 5px;
        }
        button:hover {
            background: #0f0;
            color: #000;
        }
        .output {
            white-space: pre-line;
            margin: 10px 0;
        }
        .input-group {
            margin: 10px 0;
        }
    </style>
</head>
<body>
    <div class="container">
        <div id="output" class="output"></div>
        <div id="input-area" class="input-group"></div>
    </div>

    <script>
        // Global variables to maintain state
        let currentStep = 0;
        let tutorial = false;
        let accuracytype = 0;
        let playtime = 0;
        let number_length = 0;
        let end_time = 0;
        let score = 0;
        let questioncount = 0;
        let currentResult = 0;
        let gameRunning = false;
        let gameInterval;

        function print(text) {
            document.getElementById('output').innerHTML += text + '\n';
        }

        function clearInput() {
            document.getElementById('input-area').innerHTML = '';
        }

        function createInput(placeholder, callback) {
            const inputArea = document.getElementById('input-area');
            const input = document.createElement('input');
            input.type = 'text';
            input.placeholder = placeholder;
            input.addEventListener('keypress', function(e) {
                if (e.key === 'Enter') {
                    const value = input.value;
                    callback(value);
                }
            });
            inputArea.appendChild(input);
            input.focus();
        }

        function getUsername() {
            // Browser equivalent of getenv("USER")
            return "Player";
        }

        function randMod(max) {
            return Math.floor(Math.random() * max);
        }

        function time() {
            return Math.floor(Date.now() / 1000);
        }

        function startGame() {
            const username = getUsername();
            print(`Hello ${username}!`);
            print("This is a game written for AgnoxGD.");
            print("Would you like to see a tutorial");
            print("Type 1 for no, or 0 for yes.");
            
            createInput("Enter 0 or 1", function(input) {
                const tempInput = parseInt(input);
                if (!isNaN(tempInput)) {
                    tutorial = tempInput;
                }
                
                clearInput();
                
                if (tutorial) {
                    print("OK! Lets continue.");
                } else {
                    print("Welcome to Agg! This game tests your math skills against the clock.\n\nHow to Play:\n\n1.  Set Your Playtime: First, you'll decide how long you want to play by entering a duration in seconds.\n2.  Solve Math Problems: Once the game starts, you'll be presented with a series of random math problems: addition, division, subtraction, and multiplication. Your goal is to solve as many as you can before time runs out.\n3.  Enter Your Answer: After each problem, type your answer and press Enter. The game will immediately tell you if you're correct or incorrect.\n4.  Rack Up Points: For every correct answer, you'll earn a point. Your score will be tallied at the end.\n5.  Time's Up! The game ends automatically when your chosen playtime runs out. Good luck, and have fun!");
                }
                
                askAccuracyType();
            });
        }

        function askAccuracyType() {
            print("Would you like your accuracy as a float or as an int? Type 0 for float or 1 for int.");
            createInput("Enter 0 or 1", function(input) {
                accuracytype = parseInt(input);
                clearInput();
                askPlaytime();
            });
        }

        function askPlaytime() {
            print("\nHow long would you like to play for? Answer in seconds.");
            createInput("Enter seconds", function(input) {
                playtime = parseInt(input);
                clearInput();
                askNumberLength();
            });
        }

        function askNumberLength() {
            print("Please enter maximum possible number");
            createInput("Enter max number", function(input) {
                number_length = parseInt(input);
                clearInput();
                startGameLoop();
            });
        }

        function startGameLoop() {
            const unix_time = time();
            end_time = unix_time + playtime;
            score = 0;
            questioncount = 0;
            gameRunning = true;
            
            nextQuestion();
        }

        function nextQuestion() {
            const unix_time = time();
            
            if (unix_time >= end_time) {
                endGame();
                return;
            }
            
            const num1 = randMod(number_length) + 1;
            const num2 = randMod(number_length) + 1;
            const type = randMod(4) + 1;
            let result;
            
            if (type == 1) {
                print(`Add ${num1} and ${num2}`);
                result = num1 + num2;
            } else if (type == 2) {
                print(`Divide ${num1} by ${num2}`);
                result = num1 / num2;
            } else if (type == 3) {
                print(`Subtract ${num1} from ${num2}`);
                result = num2 - num1;
            } else if (type == 4) {
                print(`Multiply ${num1} by ${num2}`);
                result = num1 * num2;
            }
            
            currentResult = result;
            questioncount++;
            
            createInput("Enter your answer", function(entered) {
                let valid = 1;
                for (let i = 0; i < entered.length; i++) {
                    const char = entered[i];
                    if (!'0123456789.'.includes(char)) {
                        valid = 0;
                        break;
                    }
                }
                
                if (valid == 1) {
                    const answer = parseFloat(entered);
                    
                    if (currentResult == answer) {
                        print("Correct!");
                        score++;
                    } else {
                        print(`Incorrect! The correct answer is ${currentResult}.`);
                    }
                } else {
                    print("Error! An invalid character was entered.");
                    endGame();
                    return;
                }
                
                clearInput();
                setTimeout(nextQuestion, 100); // Small delay to prevent blocking
            });
        }

        function endGame() {
            gameRunning = false;
            const username = getUsername();
            
            const accuracy = score / questioncount * 100;
            
            print(`Your score is ${score}!`);
            if (score >= 1) {
                print(`Great job, ${username}!`);
            } else {
                print(`Try again!, ${username}.`);
            }
            
            if (accuracytype == 1) {
                print(`You got an accuracy of ${Math.floor(accuracy)}%!`);
            } else {
                print(`You got an accuracy of ${accuracy}%!`);
            }
        }

        // Start the game when page loads
        window.onload = function() {
            startGame();
        };
    </script>
</body>
</html>
